internal/k8s/controller.go

```diff
@@ -947,6 +960,12 @@ func (c *AviController) SetupEventHandlers(k8sinfo K8sinformers) {
 			oldobj := old.(*corev1.Node)
 			node := cur.(*corev1.Node)
 			key := utils.NodeObj + "/" + node.Name
+			if lib.IsNodePortMode() {
+				if !lib.IsValidLabelOnNode(node.GetLabels(), key) {
```

Comment:
i think we need to check the labels on the old obj too and process it if the old obj had the labels, else that node won't be removed from pool servers.

Reply:
addressed